src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,158 @@
+import {Image, Platform, StyleSheet, Text} from 'react-native';
+import React, {
+    useState,
+    useCallback,
+    useImperativeHandle,
+    forwardRef, useContext,
+} from 'react';
+import Animated, {
+    useSharedValue,
+    useAnimatedStyle,
+    useAnimatedGestureHandler,
+    withSequence,
+    withDelay,
+    withTiming,
+    withSpring,
+    runOnJS,
+} from 'react-native-reanimated';
+import {PanGestureHandler} from 'react-native-gesture-handler';
+import ToastContext from "../../context/toasterContext";
+import {toastStyles} from "./ToastStyle";
+
+type ToastType =
+    'success' | 'error' | 'warning';
+
+type ToastOptions = {
+    type: ToastType;
+    text: string;
+    duration: number;
+};
+
+const Toast = forwardRef(({}, ref) => {
+    const toastTopAnimation = useSharedValue(-100);
+    const [showing, setShowing] = useState(false);
+    const [toastType, setToastType] = useState<ToastType>();
+    const [toastText, setToastText] = useState('');
+    const [toastDuration, setToastDuration] = useState(0);
+    const TOP_VALUE = Platform.OS === 'ios' ? 60 : 20;
+    const {showToast} = useContext(ToastContext);
+
+    const show = useCallback(
+        ({type, text, duration}: ToastOptions) => {
+            setShowing(true);
+            setToastType(type);
+            setToastText(text);
+            setToastDuration(duration!);
+            toastTopAnimation.value = withSequence(
+                withTiming(TOP_VALUE),
+                withDelay(
+                    duration!,
```

Comment:
It's safer to check if `duration` is present instead of using `!`. It might be a runtime crash